src/compiler/irgen.jl

```diff
@@ -4,17 +4,39 @@ function module_setup(mod::LLVM.Module)
     triple!(mod, "amdgcn-amd-amdhsa")
 
     # add debug info metadata
-    push!(metadata(mod), "llvm.module.flags",
-         MDNode([ConstantInt(Int32(1), JuliaContext()),    # llvm::Module::Error
-                 MDString("Debug Info Version"),
-                 ConstantInt(DEBUG_METADATA_VERSION(), JuliaContext())]))
+    if LLVM.version() >= v"8.0"
+    # Set Dwarf Version to 2, the DI printer will downgrade to v2 automatically,
+    # but this is technically correct and the only version supported by NVPTX
+    LLVM.flags(mod)["Dwarf Version", LLVM.API.LLVMModuleFlagBehaviorWarning] =
+        Metadata(ConstantInt(Int32(2), JuliaContext()))
```

Comment:
We probably don't want to limit DWARF to only 2 here. Haven't checked what AMDGPU supports, but debuginfo support was mimicked on what clang does for nvptx, which might be not right for AMDGPU

Reply:
Done